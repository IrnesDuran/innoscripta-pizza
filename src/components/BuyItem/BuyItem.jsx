import React from 'react';
import { connect } from 'react-redux';
import './BuyItem.styles.scss';

import {clearItemFromCart, addItem, removeItem} from '../../redux/cart/cart.actions';

const BuyItem = (props) => {


    const dollarsPrice = props.item.price*1.108
    return(
        <div className='checkout-item'>
                <span className='title capitalize'>{props.item.title}</span>       
                <span className='quantity'>
                        <div className='arrow' onClick={() => props.removeItem(props.item)}>&#10094;</div>
                            <span className='value'>{props.item.quantity}</span>
                         <div className='arrow'onClick={() => props.addItem(props.item)}>&#10095;</div>
                    </span>
                <span className='price'>{props.item.price} EUR /{dollarsPrice.toFixed(2)} USD</span>
                <div className='remove-button' onClick={()=> props.clearItem(props.item)}>&#10005;</div>
        </div>
    )};

const mapDispatchToProps =  dispatch =>({
    clearItem: item => dispatch(clearItemFromCart(item)),
    addItem: item => dispatch(addItem(item)),
    removeItem: item => dispatch(removeItem(item))
});
    


export default connect(null,mapDispatchToProps)(BuyItem);