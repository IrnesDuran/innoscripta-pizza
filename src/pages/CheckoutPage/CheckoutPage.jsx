import React, {useState} from 'react';

import { connect } from 'react-redux';

import './CheckoutPage.styles.scss';

import { clearCart } from '../../redux/cart/cart.actions';

import BuyItem from '../../components/BuyItem/BuyItem';
import ContactDetails from '../../components/ContactDetails/ContactDetails';


const CheckoutPage = (props) => {
    console.log(props)

     const [isProceeding, setIsProceeding] = useState(false);

     const delayRedirect = ( ) => {
        setIsProceeding(true);
        setTimeout(() => {
            setIsProceeding(false);
            props.history.push("/");
            props.clearCart();
        }, 3000);
     };


    const Ver1 = () => (<div className="text-6xl text-center"><h1>Thank you for your order! It will be on your address soon!</h1></div>);
 
    const deliveryPrice = props.cartItems.length ? 6 : 0;

    const total = props.cartItems.reduce((accumalatedQuantity, cartItem) => accumalatedQuantity + cartItem.quantity * cartItem.price,deliveryPrice);
    
    
    return( <div>
  { isProceeding ? <Ver1/> :
      
    
    ( <div className='checkout-page'>
        <div className='checkout-header'>
            <div className='header-block'>
                <span>Product</span>
            </div>
            <div className='header-block'>
                <span>Quantity</span>
            </div>
            <div className='header-block'>
                <span>Price</span>
            </div>
            <div className='header-block'>
                <span>Remove</span>
            </div>
        </div>
                {props.cartItems.map(item => 
                (<BuyItem key= {item.id} item={item}/>))}
                {props.cartItems.length ? <div className="ml-auto text-lg mt-2"> Delivery price:EUR ${deliveryPrice}/ USD ${deliveryPrice*1.108.toFixed(2)} </div> : null}
        <div className='total'><span>TOTAL: EUR {total.toFixed(2)}/ USD {(total*1.108).toFixed(2)}</span></div>
        <div> {props.cartItems.length ? <ContactDetails onToggle={delayRedirect} />: null} </div>
    </div>)}
    </div>
)};


const mapStateToProps=  state => {
    return {
        cartItems: state.cart.cartItems
    }
};

const mapDispatchToProps = dispatch => {
    return{
        clearCart: () => dispatch(clearCart())
    }
};



export default connect(mapStateToProps,mapDispatchToProps)(CheckoutPage);
