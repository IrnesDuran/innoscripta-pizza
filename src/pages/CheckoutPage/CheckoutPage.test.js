import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


import {CheckoutPage} from './CheckoutPage';
import BuyItem from '../../components/BuyItem/BuyItem';




describe('Testing CheckoutPage',()=>{

    let mockProps;
    let wrapper;

    beforeEach(() => {
        mockProps = {cartItems:[]}
        wrapper = shallow(<CheckoutPage {...mockProps}/>);
    });
    

 

    it('should test CheckoutPage snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    });


    it('should test if cart items exist',()=>{
        expect(wrapper.find(BuyItem)).toHaveLength(0);

    });

});
