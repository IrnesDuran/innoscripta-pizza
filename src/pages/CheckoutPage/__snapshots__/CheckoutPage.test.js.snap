// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CheckoutPage should test CheckoutPage snapshot 1`] = `ShallowWrapper {}`;
